### PR TITLE
Adding Relay testcases

### PR DESCRIPTION
**Pull request by damodaravadhani**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/relay/Azure.ResourceManager.Relay/src/Customized/RelayNamespaceData.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#nullable disable
+using System;
```
Comment (Yao725) [inline sdk/relay/Azure.ResourceManager.Relay/src/Customized/RelayNamespaceData.cs:4]: Please run `eng\scripts\Export-API.ps1 relay` in sdk directory to update the API file.

File: sdk/relay/Azure.ResourceManager.Relay/tests/Tests/PrivateEndpointConnectionTests.cs [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+            PrivateEndpointResource privateEndpoint = await CreatePrivateEndpoint();
+            Assert.AreEqual(privateEndpoint.Data.ManualPrivateLinkServiceConnections.Count, 1);
+
+            List<RelayPrivateEndpointConnectionResource> privateEndpointConnections = await _privateEndpointConnectionCollection.GetAllAsync().ToEnumerableAsync();
+            Assert.AreEqual(1, privateEndpointConnections.Count);
+            VerifyPrivateEndpointConnections(privateEndpoint.Data.ManualPrivateLinkServiceConnections[0], privateEndpointConnections[0]);
+        }
+        [Test]
+        [RecordedTest]
+        public async Task PrivateEndpointConnectionDelete()
```
Comment (Yao725) [inline sdk/relay/Azure.ResourceManager.Relay/tests/Tests/PrivateEndpointConnectionTests.cs:76]: Did you record this test? I don't see the corresponding recordings.

File: sdk/relay/Azure.ResourceManager.Relay/tests/Tests/RelayNamespaceTests.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+        [SetUp]
+        public async Task SetUp()
+        {
+            _resourceGroup = await CreateResourceGroupAsync();
+        }
+
+        [Test]
+        [RecordedTest]
+        public async Task RelayNamespaceandCreateOrUpdateTest()
```
Comment (Yao725) [inline sdk/relay/Azure.ResourceManager.Relay/tests/Tests/RelayNamespaceTests.cs:34]: Could you please double check this test? The CI complained the expected request record isn't found.

File: sdk/relay/Azure.ResourceManager.Relay/tests/Helpers/RelayTestBase.cs [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+using Azure.Core;
+using System.Security.Cryptography;
+using Azure.ResourceManager.Network;
+using Azure.ResourceManager.ManagedServiceIdentities;
+using Azure.ResourceManager.Relay.Tests;
+using Azure.ResourceManager.Relay;
+
+namespace Azure.ResourceManager.Relay.Tests.Helpers
+{
+    [ClientTestFixture]
```
Comment (Yao725) [inline sdk/relay/Azure.ResourceManager.Relay/tests/Helpers/RelayTestBase.cs:21]: I assume this attribute is needed when we want to enable the multi-version testing. Any reason that we use it here? If it's not required, I'd like to remove it.

**Review by Yao725 — COMMENTED** [review]
Please also update the CHANGELOG.md so we can know the code change that is introduced in this PR once we want to release this package.

File: sdk/relay/Azure.ResourceManager.Relay/CHANGELOG.md [hunk 5]
```diff
@@ -10,6 +10,10 @@
 
 ### Other Changes
 
+## 1.1.0 (2022-12-19)
+
+Hiding PublicNetworkAccess at namespace API as it is not currently supported.
```
Comment (Yao725) [inline sdk/relay/Azure.ResourceManager.Relay/CHANGELOG.md:15]: How about we change it to:
```suggestion
### Features Added

- Added support for PublicNetworkAccess in NetworkRuleSet API as PublicNetworkAccess at Namespace API is not currently available.
```

**Review by Yao725 — COMMENTED** [review]
- Please also update the version in Azure.ResourceManager.Relay.csproj
- Please fix the failed test in [CI](https://dev.azure.com/azure-sdk/public/_build/results?buildId=2065403&view=logs&j=ca79befc-824d-5d5b-50c5-35f083b96358&t=8393dc51-e808-5ef1-dd91-8e8b4aac1566&l=831)

**Comment by damodaravadhani** [comment]
I am not really sure why RelayNamespaceTest recording fails to record delete, for now I will remove it

**Comment by damodaravadhani** [comment]
@Yao725 , just like we use setup C# attribute on method that creates pre-requisite resources for tests to assert on. Is there any attribute that can be used to delete all resources used by tests ?

**Comment by Yao725** [comment]
@damodaravadhani, actually we will delete all the [created resource group](https://github.com/Azure/azure-sdk-for-net/blob/main/common/ManagementTestShared/Redesign/ManagementRecordedTestBase.cs#L135) after the test is over, you could also add your test clean up method and mark it as `TearDown`.

**Comment by damodaravadhani** [comment]
@Yao725, I have resolved all comments, can we merge if there aren't additional concerns ?

**Review by Yao725 — APPROVED** [review]

**Pull request merged**